### ISSUE TITLE
Does service-worker.js file need to be served with cache headers set to no-cache?

### ISSUE THREAD

**Issue by kblestarge** [GoogleChromeLabs/sw-precache#332]
Hi, I'm a PWA newb :)

I have an Angular v.4+ app built with Angular-cli. I followed [this tutorial](https://coryrylan.com/blog/fast-offline-angular-apps-with-service-workers). Using the service worker is great, except it doesn't update the application until 24 hours later after I deploy a new version of the app to production.

Here is my `sw-precache-config.js` file:
```
module.exports = {
    staticFileGlobs: [
        'latest_build/**.html',
        'latest_build/**.js',
        'latest_build/**.css',
        'latest_build/assets/img/**',
        'latest_build/assets/font/**',
        'latest_build/assets/icons/*'
    ],
    root: 'latest_build',
    stripPrefix: 'latest_build/',
    navigateFallback: '/index.html',
    runtimeCaching: [
        {
            urlPattern: /^https:\/\/xion\.store/,
            handler: 'networkFirst'
        },
        {
            urlPattern: /^https:\/\/papi\.xion\.store\/api\/v1/,
            handler: 'networkFirst'
        }
    ]
};
```

Once I create a new build of my app, I run `sw-precache --verbose --config=sw-precache-config.js` which generates a ton of new files, and I think it also generates a new `service-worker.js` file. Then, when I go to the hosted site of the app, the app does not update unless I hard refresh (CMD-SHIFT-R), but then I normal refresh and the old version is there again (until 24 hours later when it finally updates to the new version).

The browser dev tools show me this:

![screen shot 2017-10-06 at 3 45 06 pm](https://user-images.githubusercontent.com/11730298/31299856-bd5d3f8c-aaad-11e7-8657-c616117c2e78.png)

Which makes me believe the `service-worker.js` file is the culprit. It's loading from cache, until 24 hours later I think, then it's caching is expired and it fetches the new file.

So basically, my question is this:
Does the `service-worker.js` file need to be cache busted every time a new version of the app is built out?

Other information:
I am serving my app on AWS EC2 instance using [Caddy Server](https://caddyserver.com).

**Comment by midzer** [comment]
As far as i understand there is a difference between regular browser caching and the way service worker controls its files.

Browser caching is primarily set up by your web server. There is valueable information here:
https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching#defining-optimal-cache-control-policy

Service worker on the other hand, once installed, have to realize if one of their controlled files has changed. If this happens you have to prompt a small box or something like that to offer the user to reload the page, otherwise content remains old. Here is a good implementation of this behaviour: https://www.chromestatus.com/ (check out the small "toast" box coming up at page load and when content has changed)

**Comment by kblestarge** [comment]
The Angular-cli cache busts it's own files, but the service worker doesn't seem to be recognizing the changes until 24h later. Why?

**Comment by jeffposnick** [comment]
Please see https://stackoverflow.com/a/38854905/385997.

Making sure that your service worker file isn't cached is unfortunately still necessary in both Chrome and Firefox.

**Comment by kblestarge** [comment]
Can I set the cache-control max-age of the `service-worker.js` file in the sw-precache config settings? Or is this something I need to configure on my web server?

**Comment by jeffposnick** [comment]
It's a web server configuration.
